Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -590,16 +591,21 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
         do {
             try load(options: startupOptions)
             try loadVendorOptions(from: tunnelProviderProtocol)
+
+            if (try? tokenStore.fetchToken()) == nil {
+                throw TunnelError.startingTunnelWithoutAuthToken
+            }
         } catch {
             if startupOptions.startupMethod == .automaticOnDemand {
                 // If the VPN was started by on-demand without the basic prerequisites for
                 // it to work we skip firing pixels.  This should only be possible if the
                 // manual start attempt that preceded failed, or if the subscription has
                 // expired.  In either case it should be enough to record the manual failures
                 // for these prerequisited to avoid flooding our metrics.
+                providerEvents.fire(.tunnelStartOnDemandWithoutAccessToken)
```

Comment:
Just realized I missed this before but this is actually interesting on the daily too, not just count - so we can see how many users are stuck.

That said I don't want to block you any further so this is approved.